lib/classes/Service.js

```diff
@@ -111,26 +110,23 @@ class Service {
     that.plugins = serverlessFile.plugins;
     that.resources = serverlessFile.resources;
     that.functions = serverlessFile.functions || {};
+    that.configValidationMode = serverlessFile.configValidationMode || 'warn';
     that.disabledDeprecations = serverlessFile.disabledDeprecations;
 
     // merge so that the default settings are still in place and
     // won't be overwritten
     that.provider = _.merge(that.provider, serverlessFile.provider);
 
     if (serverlessFile.package) {
-      that.package.individually = serverlessFile.package.individually;
-      that.package.path = serverlessFile.package.path;
-      that.package.artifact = serverlessFile.package.artifact;
-      that.package.exclude = serverlessFile.package.exclude;
-      that.package.include = serverlessFile.package.include;
-      that.package.excludeDevDependencies = serverlessFile.package.excludeDevDependencies;
+      that.package = serverlessFile.package;
     }
 
     if (that.provider.name === 'aws') {
       that.layers = serverlessFile.layers || {};
     }
 
     that.outputs = serverlessFile.outputs;
+    that.initialServerlessFile = serverlessFile;
```

Comment:
Let's not assign it to `that` (as I see, we do not rely on this property outside of this method).

Also let's maybe let's name it `initialServerlessConfig`